--- a/src/connectivity/wlan/lib/fidl-ext/src/responder_ext.rs
+++ b/src/connectivity/wlan/lib/fidl-ext/src/responder_ext.rs
@@ -4,6 +4,7 @@
 
 use crate::{SendResultExt, TryUnpack};
 use anyhow::format_err;
+#[cfg(target_os = "fuchsia")]
 use fidl_fuchsia_wlan_softmac as fidl_softmac;
 
 /// Defines an abstract ResponderExt trait usually implemented using the impl_responder_ext!() macro.
@@ -63,6 +64,7 @@
     }
 }
 
+#[cfg(target_os = "fuchsia")]
 impl ResponderExt for fidl_fuchsia_wlan_device_service::DeviceMonitorCreateIfaceResponder {
     type Response<'a> = Result<
         &'a fidl_fuchsia_wlan_device_service::DeviceMonitorCreateIfaceResponse,
@@ -76,6 +78,7 @@
     }
 }
 
+#[cfg(target_os = "fuchsia")]
 impl ResponderExt for fidl_softmac::WlanSoftmacIfcBridgeNotifyScanCompleteResponder {
     type Response<'a> = ();
     const REQUEST_NAME: &'static str =
@@ -86,6 +89,7 @@
     }
 }
 
+#[cfg(target_os = "fuchsia")]
 impl ResponderExt for fidl_softmac::WlanSoftmacIfcBridgeReportTxResultResponder {
     type Response<'a> = ();
     const REQUEST_NAME: &'static str =
