diff -ruN a/src/connectivity/wlan/lib/sme/src/ap/mod.rs b/src/connectivity/wlan/lib/sme/src/ap/mod.rs
--- a/src/connectivity/wlan/lib/sme/src/ap/mod.rs
+++ b/src/connectivity/wlan/lib/sme/src/ap/mod.rs
@@ -516,7 +516,7 @@
         | fidl_ieee80211::WlanPhyType::He => {
             return Err(StartResult::InvalidArguments(format!("Unsupported PHY type: {phy:?}")));
         }
-        fidl_common::WlanPhyTypeUnknown!() => {
+        fidl_ieee80211::WlanPhyTypeUnknown!() => {
             return Err(StartResult::InvalidArguments(format!("Unknown PHY type: {phy:?}")));
         }
     }
diff -ruN a/src/connectivity/wlan/lib/sme/src/client/mod.rs b/src/connectivity/wlan/lib/sme/src/client/mod.rs
--- a/src/connectivity/wlan/lib/sme/src/client/mod.rs
+++ b/src/connectivity/wlan/lib/sme/src/client/mod.rs
@@ -382,10 +382,8 @@
         // Note: For association, we don't have a failure type for timeout, so cannot deduce
         //       whether an association failure is due to timeout.
         match self {
-            ConnectFailure::AuthenticationFailure(failure) => match failure {
-                fidl_ieee80211::StatusCode::RejectedSequenceTimeout => true,
-                _ => false,
-            },
+            ConnectFailure::AuthenticationFailure(failure) =>
+                failure == &fidl_ieee80211::StatusCode::RejectedSequenceTimeout,
             ConnectFailure::EstablishRsnaFailure(failure) => match failure {
                 EstablishRsnaFailure {
                     reason: EstablishRsnaFailureReason::RsnaResponseTimeout(_),
diff -ruN a/src/connectivity/wlan/lib/sme/src/client/scan.rs b/src/connectivity/wlan/lib/sme/src/client/scan.rs
--- a/src/connectivity/wlan/lib/sme/src/client/scan.rs
+++ b/src/connectivity/wlan/lib/sme/src/client/scan.rs
@@ -221,8 +221,11 @@
         security_support: &fidl_common::SecuritySupport,
     ) {
         if let Some(session) = self.scheduled_scan_receivers.get_mut(&txn_id) {
+            #[cfg(target_os = "fuchsia")]
             let bss_map = std::mem::take(&mut session.bss_map);
+            #[cfg(target_os = "fuchsia")]
             let bss_description_list = convert_bss_map(bss_map, None::<Ssid>, sme_inspect);
+            #[cfg(target_os = "fuchsia")]
             let results_fidl = bss_description_list
                 .into_iter()
                 .map(|bss_description| {
@@ -237,6 +240,7 @@
                 .map(Into::into)
                 .collect::<Vec<_>>();

+            #[cfg(target_os = "fuchsia")]
             match wlan_common::scan::write_vmo(results_fidl) {
                 Ok(vmo) => {
                     let _ = session.scan_results_sender.unbounded_send(vmo);
diff -ruN a/src/connectivity/wlan/lib/sme/src/lib.rs b/src/connectivity/wlan/lib/sme/src/lib.rs
--- a/src/connectivity/wlan/lib/sme/src/lib.rs
+++ b/src/connectivity/wlan/lib/sme/src/lib.rs
@@ -16,6 +16,7 @@

 pub mod ap;
 pub mod client;
+#[cfg(feature = "fuchsia-transport")]
 pub mod serve;
 #[cfg(test)]
 pub mod test_utils;
