--- a/src/connectivity/network/dhcpv4/client/core/src/deps.rs
+++ b/src/connectivity/network/dhcpv4/client/core/src/deps.rs
@@ -7,6 +7,7 @@
 //! purposes.
 
 use diagnostics_traits::InspectableInstant;
+#[cfg(target_os = "fuchsia")]
 use fuchsia_async as fasync;
 use rand::Rng;
 use std::future::Future;
@@ -118,6 +119,7 @@
     fn average(&self, other: Self) -> Self;
 }
 
+#[cfg(target_os = "fuchsia")]
 impl Instant for fasync::MonotonicInstant {
     fn add(&self, duration: std::time::Duration) -> Self {
         *self + duration.into()
