diff -ruN a/src/connectivity/wlan/wlancfg/src/client/types.rs b/src/connectivity/wlan/wlancfg/src/client/types.rs
--- a/src/connectivity/wlan/wlancfg/src/client/types.rs	2026-08-09 22:11:17.637528745 +0000
+++ b/src/connectivity/wlan/wlancfg/src/client/types.rs	2026-08-09 22:11:17.659700262 +0000
@@ -5,6 +5,9 @@
 use crate::config_management::{self};
 use crate::util::historical_list::Timestamped;
 use fidl_fuchsia_wlan_internal as fidl_internal;
+#[cfg(feature = "host-selection")]
+use crate::fidl_fuchsia_wlan_policy as fidl_policy;
+#[cfg(not(feature = "host-selection"))]
 use fidl_fuchsia_wlan_policy as fidl_policy;
 use fidl_fuchsia_wlan_sme as fidl_sme;
 use fuchsia_async::MonotonicInstant;
@@ -12,6 +15,12 @@
 use wlan_common::channel::Channel;
 use wlan_common::security::SecurityAuthenticator;
 use wlan_common::sequestered::Sequestered;
+#[cfg(feature = "host-selection")]
+use crate::wlan_metrics_registry::{
+    PolicyConnectionAttemptMigratedMetricDimensionReason,
+    PolicyDisconnectionMigratedMetricDimensionReason,
+};
+#[cfg(not(feature = "host-selection"))]
 use wlan_metrics_registry::{
     PolicyConnectionAttemptMigratedMetricDimensionReason,
     PolicyDisconnectionMigratedMetricDimensionReason,
diff -ruN a/src/connectivity/wlan/wlancfg/src/config_management/config_manager.rs b/src/connectivity/wlan/wlancfg/src/config_management/config_manager.rs
--- a/src/connectivity/wlan/wlancfg/src/config_management/config_manager.rs	2026-08-09 22:11:17.622551284 +0000
+++ b/src/connectivity/wlan/wlancfg/src/config_management/config_manager.rs	2026-08-09 22:11:17.659264225 +0000
@@ -7,6 +7,7 @@
     NetworkConfig, NetworkConfigError, NetworkIdentifier, PastConnectionData, PastConnectionList,
     SecurityType,
 };
+#[cfg(not(feature = "host-selection"))]
 use super::stash_conversion::*;
 use crate::client::types::{self, ScanObservation};
 use crate::telemetry::{TelemetryEvent, TelemetrySender};
@@ -18,6 +19,7 @@
 use futures::lock::Mutex;
 use log::{error, info};
 use std::collections::{HashMap, HashSet};
+#[cfg(not(feature = "host-selection"))]
 use wlan_storage::policy::{POLICY_STORAGE_ID, PolicyStorage};
 
 const MAX_SAVED_NETWORKS: usize = 1000;
@@ -27,6 +29,7 @@
 /// (SSID and security protocol). Network configs are saved in-memory, and part of each network
 /// data is saved persistently. Futures aware locks are used in order to wait for the storage flush
 /// operations to complete when data changes.
+#[cfg(not(feature = "host-selection"))]
 pub struct SavedNetworksManager {
     saved_networks: Mutex<NetworkConfigMap>,
     // Persistent storage for networks, which should be updated when there is a change to the data
@@ -118,6 +121,7 @@
     ) -> PastConnectionList;
 }
 
+#[cfg(not(feature = "host-selection"))]
 impl SavedNetworksManager {
     /// Initializes a new Saved Network Manager by reading saved networks from local storage using
     /// a WLAN helper library. It will attempt to migrate any data from legacy storage.
@@ -201,6 +205,7 @@
     }
 }
 
+#[cfg(not(feature = "host-selection"))]
 #[async_trait(?Send)]
 impl SavedNetworksManagerApi for SavedNetworksManager {
     async fn remove(&self, network_id: NetworkIdentifier) -> Result<bool, NetworkConfigError> {
diff -ruN a/src/connectivity/wlan/wlancfg/src/config_management/network_config.rs b/src/connectivity/wlan/wlancfg/src/config_management/network_config.rs
--- a/src/connectivity/wlan/wlancfg/src/config_management/network_config.rs	2026-08-09 22:11:17.630665577 +0000
+++ b/src/connectivity/wlan/wlancfg/src/config_management/network_config.rs	2026-08-09 22:11:17.659530086 +0000
@@ -7,6 +7,9 @@
 use arbitrary::Arbitrary;
 #[cfg(test)]
 use fidl_fuchsia_wlan_internal as fidl_internal;
+#[cfg(feature = "host-selection")]
+use crate::fidl_fuchsia_wlan_policy as fidl_policy;
+#[cfg(not(feature = "host-selection"))]
 use fidl_fuchsia_wlan_policy as fidl_policy;
 use fuchsia_async as fasync;
 use log::info;
diff -ruN a/src/lib/diagnostics/inspect/contrib/rust/host/src/lib.rs b/src/lib/diagnostics/inspect/contrib/rust/host/src/lib.rs
--- a/src/lib/diagnostics/inspect/contrib/rust/host/src/lib.rs	2026-08-09 22:11:17.644255998 +0000
+++ b/src/lib/diagnostics/inspect/contrib/rust/host/src/lib.rs	2026-08-09 22:11:17.659804870 +0000
@@ -16,6 +16,15 @@
 pub mod log {
     use std::marker::PhantomData;
 
+    pub trait WriteInspect {
+        fn write_inspect<'a>(
+            &self,
+            writer: &fuchsia_inspect::Node,
+            key: impl Into<std::borrow::Cow<'a, str>>,
+        );
+    }
+
+    pub struct InspectList<'a, T>(pub &'a [T]);
     pub struct InspectBytes<'a>(pub &'a [u8]);
     pub struct InspectUintArray<'a>(pub &'a [u8]);
     pub struct InspectListClosure<'a, T, F>(pub &'a [T], pub F);
--- a/src/connectivity/wlan/wlancfg/src/client/scan/mod.rs
+++ b/src/connectivity/wlan/wlancfg/src/client/scan/mod.rs
@@ -27,6 +27,9 @@
 
 mod fidl_conversion;
 mod queue;
+mod selection_conversion;
+
+use selection_conversion::{bss_to_network_map, network_map_to_scan_result};
 
 pub use fidl_conversion::{
     scan_result_to_policy_scan_result, send_scan_error_over_fidl, send_scan_results_over_fidl,
@@ -361,83 +364,6 @@
     }
 }
 
-/// Converts sme::ScanResult to our internal BSS type, then adds it to a map.
-/// Only keeps the first unique instance of a BSSID
-fn bss_to_network_map(
-    scan_result_list: Vec<wlan_common::scan::ScanResult>,
-    target_ssids: &[types::Ssid],
-    scan_event_inspect_data: &mut ScanEventInspectData,
-) -> HashMap<types::NetworkIdentifierDetailed, Vec<types::Bss>> {
-    let mut bss_by_network: HashMap<types::NetworkIdentifierDetailed, Vec<types::Bss>> =
-        HashMap::new();
-    for scan_result in scan_result_list.into_iter() {
-        let security_type: types::SecurityTypeDetailed =
-            scan_result.bss_description.protection().into();
-        if security_type == types::SecurityTypeDetailed::Unknown {
-            // Log a space-efficient version of the IEs.
-            let readable_ie =
-                scan_result.bss_description.ies().iter().map(|n| n.to_string()).join(",");
-            debug!("Encountered unknown protection, ies: [{:?}]", readable_ie.clone());
-            scan_event_inspect_data.unknown_protection_ies.push(readable_ie);
-        };
-        let entry = bss_by_network
-            .entry(types::NetworkIdentifierDetailed {
-                ssid: scan_result.bss_description.ssid.clone(),
-                security_type,
-            })
-            .or_default();
-
-        // Check if this BSSID is already in the hashmap
-        if !entry.iter().any(|existing_bss| existing_bss.bssid == scan_result.bss_description.bssid)
-        {
-            entry.push(types::Bss {
-                bssid: scan_result.bss_description.bssid,
-                signal: types::Signal {
-                    rssi_dbm: scan_result.bss_description.rssi_dbm,
-                    snr_db: scan_result.bss_description.snr_db,
-                },
-                channel: scan_result.bss_description.channel,
-                timestamp: scan_result.timestamp,
-                // TODO(123709): if target_ssids contains the wildcard, this need to be "Unknown"
-                observation: if target_ssids.contains(&scan_result.bss_description.ssid) {
-                    types::ScanObservation::Active
-                } else {
-                    types::ScanObservation::Passive
-                },
-                compatibility: scan_result.compatibility,
-                bss_description: wlan_common::sequestered::Sequestered::from(
-                    fidl_fuchsia_wlan_ieee80211::BssDescription::from(scan_result.bss_description),
-                ),
-            });
-        };
-    }
-    bss_by_network
-}
-
-fn network_map_to_scan_result(
-    mut bss_by_network: HashMap<types::NetworkIdentifierDetailed, Vec<types::Bss>>,
-) -> Vec<types::ScanResult> {
-    let mut scan_results: Vec<types::ScanResult> = bss_by_network
-        .drain()
-        .map(|(types::NetworkIdentifierDetailed { ssid, security_type }, bss_entries)| {
-            let compatibility = if bss_entries.iter().any(|bss| bss.is_compatible()) {
-                fidl_policy::Compatibility::Supported
-            } else {
-                fidl_policy::Compatibility::DisallowedNotSupported
-            };
-            types::ScanResult {
-                ssid,
-                security_type_detailed: security_type,
-                entries: bss_entries,
-                compatibility,
-            }
-        })
-        .collect();
-
-    scan_results.sort_by(|a, b| a.ssid.cmp(&b.ssid));
-    scan_results
-}
-
 fn log_metric_for_scan_error(reason: &fidl_sme::ScanErrorCode, scan_defects: &mut Vec<ScanIssue>) {
     let metric_type = match *reason {
         fidl_sme::ScanErrorCode::NotSupported
--- a/src/connectivity/wlan/wlancfg/src/client/state_machine.rs
+++ b/src/connectivity/wlan/wlancfg/src/client/state_machine.rs
@@ -20,8 +20,15 @@
 use anyhow::format_err;
 use fidl_fuchsia_wlan_ieee80211 as fidl_ieee80211;
 use fidl_fuchsia_wlan_internal as fidl_internal;
+#[cfg(feature = "host-selection")]
+use crate::fidl_fuchsia_wlan_policy as fidl_policy;
+#[cfg(not(feature = "host-selection"))]
 use fidl_fuchsia_wlan_policy as fidl_policy;
 use fidl_fuchsia_wlan_sme as fidl_sme;
+#[cfg(feature = "host-selection")]
+use crate::mode_management::{ClientSmeEventStream, ConnectTransactionEventStream};
+#[cfg(not(feature = "host-selection"))]
+use fidl_sme::{ClientSmeEventStream, ConnectTransactionEventStream};
 use fuchsia_async::{self as fasync, DurationExt};
 use fuchsia_inspect::Node as InspectNode;
 use fuchsia_inspect_contrib::inspect_insert;
@@ -167,7 +174,7 @@
 pub async fn serve(
     iface_id: u16,
     proxy: SmeForClientStateMachine,
-    sme_event_stream: fidl_sme::ClientSmeEventStream,
+    sme_event_stream: ClientSmeEventStream,
     req_stream: mpsc::Receiver<ManualRequest>,
     update_sender: ClientListenerMessageSender,
     saved_networks_manager: Arc<dyn SavedNetworksManagerApi>,
@@ -575,7 +582,7 @@
     network_identifier: types::NetworkIdentifier,
     credential: Credential,
     ess_connect_reason: types::ConnectReason,
-    connect_txn_stream: fidl_sme::ConnectTransactionEventStream,
+    connect_txn_stream: ConnectTransactionEventStream,
     network_is_likely_hidden: bool,
     ess_connect_start_time: fasync::MonotonicInstant,
     bss_connect_start_time: fasync::MonotonicInstant,
@@ -596,7 +603,7 @@
         network_identifier: types::NetworkIdentifier,
         credential: Credential,
         ess_connect_reason: types::ConnectReason,
-        connect_txn_stream: fidl_sme::ConnectTransactionEventStream,
+        connect_txn_stream: ConnectTransactionEventStream,
         network_is_likely_hidden: bool,
     ) -> Self {
         // Tracked signals
