--- a/src/connectivity/wlan/lib/common/rust/src/error.rs
+++ b/src/connectivity/wlan/lib/common/rust/src/error.rs
@@ -21,6 +21,7 @@ impl From<append::BufferTooSmall> for FrameWriteError {
     }
 }
 
+#[cfg(target_os = "fuchsia")]
 impl From<zx::Status> for FrameWriteError {
     fn from(status: zx::Status) -> Self {
         FrameWriteError::BadWrite(status.to_string())
--- a/src/connectivity/wlan/lib/common/rust/src/scan.rs
+++ b/src/connectivity/wlan/lib/common/rust/src/scan.rs
@@ -357,6 +357,8 @@ impl TryFrom<fidl_sme::ScanResult> for ScanResult {
 
     fn try_from(scan_result: fidl_sme::ScanResult) -> Result<ScanResult, Self::Error> {
         let fidl_sme::ScanResult { compatibility, timestamp_nanos, bss_description } = scan_result;
+        #[cfg(not(target_os = "fuchsia"))]
+        let _ = timestamp_nanos;
         Ok(ScanResult {
             compatibility: Compatibility::try_from_fidl(compatibility)
                 .map_err(|_| format_err!("failed to convert FIDL `Compatibility`"))?,
