--- a/src/connectivity/wlan/lib/common/rust/src/error.rs
+++ b/src/connectivity/wlan/lib/common/rust/src/error.rs
@@ -22,6 +22,7 @@
     }
 }
 
+#[cfg(target_os = "fuchsia")]
 impl From<zx::Status> for FrameWriteError {
     fn from(status: zx::Status) -> Self {
         FrameWriteError::BadWrite(status.to_string())
--- a/src/connectivity/wlan/lib/common/rust/src/scan.rs
+++ b/src/connectivity/wlan/lib/common/rust/src/scan.rs
@@ -357,6 +357,8 @@
 
     fn try_from(scan_result: fidl_sme::ScanResult) -> Result<ScanResult, Self::Error> {
         let fidl_sme::ScanResult { compatibility, timestamp_nanos, bss_description } = scan_result;
+        #[cfg(not(target_os = "fuchsia"))]
+        let _ = timestamp_nanos;
         Ok(ScanResult {
             compatibility: Compatibility::try_from_fidl(compatibility)
                 .map_err(|_| format_err!("failed to convert FIDL `Compatibility`"))?,
--- a/src/connectivity/wlan/lib/common/rust/src/lib.rs
+++ b/src/connectivity/wlan/lib/common/rust/src/lib.rs
@@ -25,7 +25,7 @@
 pub mod sequestered;
 pub mod sink;
 pub mod stats;
-#[cfg(target_os = "fuchsia")]
+#[cfg(any(target_os = "fuchsia", feature = "host_test_utils"))]
 pub mod test_utils;
 pub mod tim;
 pub mod time;
--- a/src/connectivity/wlan/lib/common/rust/src/time.rs
+++ b/src/connectivity/wlan/lib/common/rust/src/time.rs
@@ -26,9 +26,7 @@
     Hash,
 )]
 pub struct TimeUnit(pub u16);
 
-#[cfg(target_os = "fuchsia")]
-#[cfg(target_os = "fuchsia")]
 impl From<TimeUnit> for zx::MonotonicDuration {
     fn from(tu: TimeUnit) -> zx::MonotonicDuration {
         zx::MonotonicDuration::from_micros(tu.into_micros())
--- a/src/connectivity/wlan/lib/common/rust/src/test_utils/mod.rs
+++ b/src/connectivity/wlan/lib/common/rust/src/test_utils/mod.rs
@@ -3,16 +3,23 @@
 // found in the LICENSE file.
 
 use crate::append::{Append, BufferTooSmall, TrackedAppend, VecCursor};
+#[cfg(target_os = "fuchsia")]
 use fuchsia_async::{DurationExt, OnTimeout, TimeoutExt};
 
+#[cfg(target_os = "fuchsia")]
 use futures::Future;
 
+#[cfg(target_os = "fuchsia")]
 pub mod fake_capabilities;
+#[cfg(target_os = "fuchsia")]
 pub mod fake_features;
+#[cfg(target_os = "fuchsia")]
 pub mod fake_frames;
+#[cfg(target_os = "fuchsia")]
 pub mod fake_stas;
 
 /// A trait which allows to expect a future to terminate within a given time or panic otherwise.
+#[cfg(target_os = "fuchsia")]
 pub trait ExpectWithin: Future + Sized {
     fn expect_within<S: ToString + Clone>(
         self,
@@ -24,6 +31,7 @@
     }
 }
 
+#[cfg(target_os = "fuchsia")]
 impl<F: Future + Sized> ExpectWithin for F {}
 
 pub struct FixedSizedTestBuffer(VecCursor);
